compiler/src/iree/compiler/Codegen/Common/FissionTransferOpsInControlFlow.cpp

```diff
@@ -267,7 +271,12 @@ struct FissionTransferOpsInControlFlowPass final
       if (failed(result)) {
         continue;
       }
-      fissionTargets.push_back(result.value());
+      FissionTarget currentTarget = result.value();
+      if (!FissionMultiTrip &&
+          !neverRunsSecondIteration(currentTarget.parent)) {
+        return;
+      }
```

Comment:
I think  it is cleaner to use forOp instead of `currentTarget` because they are guaranteed to be equivalent. Also use continue instead of return so fission can continue to operate on the next forOp if the current loop doesn't satisfy.